keep/contextmanager/contextmanager.py

```diff
@@ -147,6 +145,10 @@ def set_condition_results(
                 **kwargs,
             }
         )
+        # update the current for each context
+        self.foreach_context.update(
+            {"compare_value": compare_value, "compare_to": compare_to, **kwargs}
+        )
```

Comment:
i'm still not sure how foreach works, tbh.. what's the foreach_context? is it the CURRENT foreach iteration?

Reply:
yes. this is the way of extending `foreach` with other attributes (e.g. `level` or `stddev`), see `foreach.md`

Reply:
@talboren can mark as resolved?